core/modules/wcontrol/TransmitMgr.h

```diff
@@ -62,51 +62,72 @@ class TransmitMgr {
     TransmitMgr& operator=(TransmitMgr const&) = delete;
     virtual ~TransmitMgr() = default;
 
-    int getTotalCount() { return _totalCount; }
-    int getTransmitCount() { return _transmitCount; }
-    int getAlreadyTransCount() { return _alreadyTransCount; }
+    int getTotalCount(int czarId);
+    int getTransmitCount(int czarId);
+    int getAlreadyTransCount(int czarId);
 
-    virtual std::ostream& dump(std::ostream &os) const;
-    std::string dump() const;
+    /// Class methods, that have already locked '_mtx', should call 'dumpBase'.
+    std::ostream& dump(std::ostream &os) const;
+
+    /// This will try to lock 'TransmitMgr::_mtx'.
     friend std::ostream& operator<<(std::ostream &out, TransmitMgr const& mgr);
 
     friend class TransmitLock;
 
+protected:
+    /// _mtx must be locked before calling this function.
+    /// Dump the contents of the class to a string for logging.
+    virtual std::ostream& dumpBase(std::ostream &os) const;
+
+    /// _mtx must be locked before calling this function.
+    std::string dump() const;
+
 private:
-    void _take(bool interactive, bool alreadyTransmitting);
+    void _take(bool interactive, bool alreadyTransmitting, int czarId);
+
+    void _release(bool interactive, bool alreadyTransmitting, int czarId);
+
+    /// This class is used to store transmit information for a czar
+    class TransmitInfo {
```

Comment:
Since this is a private class within the class `TransmitMgr` then you may make all member `public` and avoid things like `friend class TransmitMgr;`. You may also drop all constructors because you are initializing counters to zeroes.